src/foremast/utils/dns.py

```diff
@@ -104,14 +105,14 @@ def delete_existing_cname(env, zone_id, dns_name):
     startrecord = None
     newrecord_name = dns_name
     client = boto3.Session(profile_name=env).client('route53')
-    response = client.list_resource_record_sets(
-        HostedZoneId=zone_id
-    )
-    for item in response['ResourceRecordSets']:
-        if item['Name'].rstrip('.') == newrecord_name and item['Type'] == 'CNAME':
-            startrecord = item
-            LOG.info("Found old record: %s", item)
-            break
+    pager = client.get_paginator('list_resource_record_sets')
+
+    for rset in pager.paginate(HostedZoneId=zone_id):
+        for record in rset['ResourceRecordSets']:
+            if record['Name'].rstrip('.') == newrecord_name and record['Type'] == 'CNAME':
+                startrecord = record
+                LOG.info("Found old record: %s", record)
+                break
```

Comment:
This block has minor differences from `primary_record_exists()`. Maybe make that function slightly more generic and use that here.